juniper_official/Kernel/krt-statistics-kpis.playbook

```diff
@@ -0,0 +1,159 @@
+/*
+ * This playbook is used to detect KRT Async queue stuck condition and raise alarm.
+ * 
+ * Following are the set of commands and the corresponding RPCs used to detect the problematic state.
+ * 
+ * labroot@jtac-mx960> show krt state | display xml rpc
+ * <rpc-reply xmlns:junos="http://xml.juniper.net/junos/18.2I0/junos">
+ *     <rpc>
+ *         <get-krt-state>
+ *         </get-krt-state>
+ *     </rpc>
+ *     <cli>
+ *         <banner></banner>
+ *     </cli>
+ * </rpc-reply>
+ * 
+ * labroot@jtac-ptx10001-20c-proto-r2001> show krt async io statistics | display xml rpc
+ * <rpc-reply xmlns:junos="http://xml.juniper.net/junos/18.2I0/junos">
+ *     <rpc>
+ *         <get-async-io-stats>
+ *         </get-async-io-stats>
+ *     </rpc>
+ *     <cli>
+ *         <banner></banner>
+ *     </cli>
+ * </rpc-reply>
+ * 
+ * labroot@jtac-ptx10001-20c-proto-r2001> show krt async correlator statistics | display xml rpc
+ * <rpc-reply xmlns:junos="http://xml.juniper.net/junos/18.2I0/junos">
+ *     <rpc>
+ *         <get-async-cor-stats>
+ *         </get-async-cor-stats>
+ *     </rpc>
+ *     <cli>
+ *         <banner></banner>
+ *     </cli>
+ * </rpc-reply>
+ * 
+ * labroot@jtac-mx960> show krt state | display xml
```

Comment:
Nice.. I like this view of the device-output, though I can't decide if this should be here in the playbook file or in the rule file. 
When we introduce the playbook-abstraction feature, the goal is to allow a playbook to have #ifdef compilations/logic built in for rules depending on the meta-data properties. So, for instance, I can have a single interface-KPIs playbook pull in OC rules for MX >= 17.1; iAgent rules for MX< 17.1 && all SRX, etc..

Reply:
Agree with you